run/radius2john.py

```diff
@@ -1,68 +1,163 @@
+        key = f"{ip.dst}-{radius_packet.id}"
+        if key in all_requests:
+            dump_response(args, ip.dst, all_requests[key], radius_packet.authenticator, udpdata)
+
+
+def dump_response(args, ip, req_ra, ra, udpdata):  # 3.1 attack
+    if args["single"] and ip in dumped_ips:
+        return
+
+    salt = bytearray(udpdata)
+    salt[4:20] = req_ra  # Replace Response Authenticator with the Request Authenticator
+
+    response_type = "1009" if len(salt) <= 16 else "1017"
+    print(
+        f"{ip}:$dynamic_{response_type}${binascii.hexlify(ra).decode()}$HEX${binascii.hexlify(salt).decode('utf-8')}"
+    )
+
+    dumped_ips[ip] = "reply"
+
+
+def dump_access_request(args, ip, ra, hashed):  # 3.3 attack
+    if args["single"] and ip in dumped_ips and dumped_ips[ip] == "request":
+        return
+
+    xor_result = bytes(a ^ b for a, b in zip(hashed, args['password'][:16].encode('utf-8').ljust(16, b'\x00')))
+
+    print(
+        f"{ip}:$dynamic_1008${binascii.hexlify(xor_result).decode()}$HEX${binascii.hexlify(ra).decode('utf-8')}"
+    )
 
-from binascii import hexlify
-from sys import argv
-import os
+    dumped_ips[ip] = "request"
 
-if len(argv) < 2:
-    print('Usage: ./radius2john.py <pcap files>')
-    exit(1)
 
-filenames = argv[1:]
+if __name__ == "__main__":
 
-for filename in filenames:
+    parser = argparse.ArgumentParser(formatter_class=argparse.RawDescriptionHelpFormatter,epilog=
+    """
+    ### Utility to bruteforce RADIUS shared-secret written by k4amos
+    Basic Usage: ./radius2john.py <pcap files>"
 
-    capture_file = rdpcap(filename)
+    ---
 
-    accounting_request_packets = {}
-    accounting_response_packets = {}
+    Application of two  methods described in http://www.untruth.org/~josh/security/radius/radius-auth.html :
+    - "3.3 User-Password Attribute Based Shared Secret Attack"
+    - "3.1 Response Authenticator Based Shared Secret Attack"
 
-    for packet in capture_file:
+    # For attack 3.3 :
+    We try authentications using a known password, and sniff the radius packets to a pcap file.
+    This script reads access-request in the pcap file, and dumps the md5(RA+secret) and RA, in a john-friendly format.
 
-        if Radius in packet:
-            src_ip = packet[IP].src
-            dst_ip = packet[IP].dst
+    # For attack 3.1:
+    We don't need to try authentications. Just sniff the radius packets in a pcap file.
+    This script reads the pcap file, matches radius responses with the corresponding all_requests,
+    and dumps md5 and salt as needed.
+    		""")
 
-            if packet[Radius].code == 4 and not (src_ip, dst_ip) in accounting_request_packets:
-                accounting_request_packets[(src_ip, dst_ip)] = packet
-            elif packet[Radius].code == 5 and not (dst_ip, src_ip) in accounting_response_packets:
-                accounting_response_packets[(dst_ip, src_ip)] = packet
+    parser.add_argument('-f', '--file', type=str, required=True, nargs='+')
```

Comment:
the `-f` option conflicts with the your hand-written usage message (at multiple places) that the basic usage is `./radius2john.py <pcap files>`. Either adapt the usage message, or better, use a positional argument without option flag for the files.